File: app/src/main/java/com/example/resourcesapp/ThemePreferenceUtil.java
package com.example.resourcesapp;

import android.content.Context;
import android.content.SharedPreferences;

import androidx.preference.PreferenceManager;

public class ThemePreferenceUtil {

    private static final String THEME_PREFERENCE_KEY = "theme_preference";
    private static final String DEFAULT_THEME = "light";

    private static SharedPreferences getSharedPreferences(Context context) {
        return PreferenceManager.getDefaultSharedPreferences(context);
    }

    public static void saveThemePreference(Context context, boolean isDarkModeEnabled) {
        SharedPreferences.Editor editor = getSharedPreferences(context).edit();
        editor.putBoolean(THEME_PREFERENCE_KEY, isDarkModeEnabled);
        editor.apply();
    }

    public static boolean getSavedThemePreference(Context context) {
        return getSharedPreferences(context).getBoolean(THEME_PREFERENCE_KEY, false);
    }

}

File: app/src/main/java/com/example/resourcesapp/ThemePreferenceUtil.kt
package com.example.resourcesapp

import android.content.Context
import android.content.SharedPreferences
import androidx.preference.PreferenceManager

object ThemePreferenceUtil {
    private const val THEME_PREFERENCE_KEY = "theme_preference"
    private const val DEFAULT_THEME = "light"
    private fun getSharedPreferences(context: Context): SharedPreferences {
        return PreferenceManager.getDefaultSharedPreferences(context)
    }

    @JvmStatic
    fun saveThemePreference(context: Context, isDarkModeEnabled: Boolean) {
        val editor = getSharedPreferences(context).edit()
        editor.putBoolean(THEME_PREFERENCE_KEY, isDarkModeEnabled)
        editor.apply()
    }

    @JvmStatic
    fun getSavedThemePreference(context: Context): Boolean {
        return getSharedPreferences(context).getBoolean(THEME_PREFERENCE_KEY, false)
    }
}